source/ui/BioXAS/BioXASCarbonFilterFarmView.cpp

```diff
@@ -0,0 +1,48 @@
+#include "BioXASCarbonFilterFarmView.h"
+#include "ui/beamline/AMExtendedControlEditor.h"
+
+BioXASCarbonFilterFarmView::BioXASCarbonFilterFarmView(BioXASCarbonFilterFarmControl *filterFarm, QWidget *parent) :
+    QWidget(parent)
+{
+	// Initialize member variables.
+
+	filterFarm_ = 0;
+
+	// Current settings.
+
+	setFilterFarm(filterFarm);
+}
+
+BioXASCarbonFilterFarmView::~BioXASCarbonFilterFarmView()
+{
+
+}
+
+void BioXASCarbonFilterFarmView::setFilterFarm(BioXASCarbonFilterFarmControl *newFilterFarm)
+{
+	if (filterFarm_ != newFilterFarm) {
+
+		if (filterFarm_) {
```

Comment:
Are you using this to delete the old editor before making the new one and then just remaking the layout with a new editor inside it?  If so, I'd recommend making the filterFarmEditor and making it a member as well as the layout.  Then you could just do,

```
farmLayout_->removeWidget(filterFarm_);
filterFarm_->deleteLater();
filterFarm_ = newFilterFarm;
farmLayout_->addWidget(filterFarm_);
```